packages/InteractiveProfilingTool-Tests.package/IPTTreeModelTestCase.class/instance/testBrowseClassMethod.st

```diff
@@ -0,0 +1,16 @@
+tests
+testBrowseClassMethod
+	
+	| tally browser tree tallyForClassMethodCall |
+	
+	tally := IPTTreeModelMock constructMessageTallyFrom: [IPTTreeModelMock computeSomething].
+	tallyForClassMethodCall := tally receivers first.
+	
+	tree := IPTTreeModelMock new 
+		currentSelection: tallyForClassMethodCall.
+	browser := tree browseMethodCall.
+	
+	[self assert: browser selectedMessageName = #constructMessageTallyFrom:. 
+	self assert: browser selectedClassName = #IPTTreeModelMock] 
+		ensure: [browser dependents first delete]
```

Comment:
Could you explain to me what this does? Tried to understand it, but I guess I’m lost 🙃

Reply:
Do you mean all three lines or just line 15? Line 15 closes any opened Browser windows at the end of the test. The two lines above test that the correct method (first line) and class (second line) are opened in the browser.

Reply:
Thanks! In this case, why can’t we just have smth like the following?

```smalltalk
self assert: browser selectedMessageName = #constructMessageTallyFrom:. 
self assert: browser selectedClassName = #IPTTreeModelMock.
	
browser dependents first delete
```

Reply:
The `ensure` acts as a finally block that ensures that the browser window is always closed. 
If we do it like you suggested, if one of the asserts fails, the browser window will stay open.